onnxruntime/test/ep_graph/test_ep_graph.cc

```diff
@@ -178,6 +178,77 @@ TEST(EpGraphTest, SerializeToProto_Mnist) {
   EXPECT_EQ(output_serialized, output_original);
 }
 
+// Test serializing an OrtGraph (MNIST) to GraphProto. Initializers are configured as "external" but point to
+// existing data in memory (not standard ONNX).
+TEST(EpGraphTest, SerializeToProto_ExternalInitializersInMemory) {
+  const ORTCHAR_T* original_model_path = ORT_TSTR("testdata/mnist.onnx");
+  auto test_graph = TestGraph::Load(original_model_path);
+  ASSERT_NE(test_graph, nullptr) << "Failed to load test model";
+
+  const OrtGraph& ort_graph = test_graph->GetOrtGraph();
+
+  auto handle_initializer_data = [](const OrtValueInfo* value_info,
+                                    const void* data, size_t bytes,
+                                    bool& is_external, std::string& location,
+                                    int64_t& offset) -> Ort::Status {
+    (void)value_info;
+    (void)bytes;
+
+    offset = reinterpret_cast<int64_t>(data);
+    location = "_MEM_ADDR_";
```

Comment:
>= "_MEM_ADDR_"; [](http://example.com/codeflow?start=13&length=15)

Try using a constant that is actually a little different from this string

Reply:
Refer to https://github.com/microsoft/onnxruntime/pull/25411#discussion_r2211714611

Reply:
I'm not sure I understand why this is necessary. This unit test is an example of how an EP could use these utilities. An EP implementer would select their own constant anyway that is different than what ORT uses internally.